pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/AnyScript.groovy

```diff
@@ -37,15 +36,11 @@ public class AnyScript extends DeclarativeAgentScript<Any> {
 
     @Override
     public Closure run(Closure body) {
-        return {
-            try {
-                script.node {
-                    body.call()
-                }
-            } catch (Exception e) {
-                script.getProperty("currentBuild").result = Result.FAILURE
-                throw e
-            }
+        Label l = (Label) Label.DescriptorImpl.instanceForName("label", [label: null])
```

Comment:
It looks overly abusive having to do `setContext` over and over again for the caller, can't that be a part of `instanceForName` ?